### ISSUE TITLE
[Bug]: Cloudflare examples action redirect returns 500

### ISSUE THREAD

**Issue by tony-maynard** [remix-run/remix#553]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [X] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.0.4

### Steps to Reproduce

1. Follow the readme to set up @remix-run/cloudflare-workers
2. Navigate to demos/actions
3. Submit the top secret answer

### Expected Behavior

App is supposed to redirect to demo/correct

### Actual Behavior

Recieve a 500 error with message from the server:

```
TypeError
Response with null body status cannot have body
```
Also returns the html as a string (Not sure if this is expected)
<img width="871" alt="Screenshot 2021-11-23 at 21 48 02" src="https://user-images.githubusercontent.com/85884885/143134538-05d42d2c-4e7c-415f-b641-8fc1f7d27b6e.png">
<img width="785" alt="Screenshot 2021-11-23 at 21 47 57" src="https://user-images.githubusercontent.com/85884885/143134531-fe57c619-4d5a-423a-984c-bca5d8145ec6.png">

**Comment by bluebeel** [comment]
Hello,

I had the same issue and after some investigation I found the problem.
When there is a redirect Remix does this piece of code.

https://github.com/remix-run/remix/blob/acd3e29990e9e096a73e7dbfd2eb03c6fe2e4dca/packages/remix-server-runtime/server.ts#L225-L237

The logic of the code is correct, however there is a small error which is the body of the response. The body is an empty string while the status code is 204.
A 204 response cannot have a body normally ([RFC](https://www.rfc-editor.org/rfc/rfc7231#section-6.3.5)).
Locally if you use `miniflare` it crashes. However when you deploy on cloudflare workers it works because they accept (for the moment) a 204 with a body. We have the following warning:

> Constructing a Response with a null body status (204) and a non-null, zero-length body. This is technically incorrect, and we recommend you update your code to explicitly pass in a `null` body, e.g. `new Response(null, { status: 204, ... })`. (We continue to allow the zero-length body behavior because it was previously the only way to construct a Response with a null body status. This behavior may change in the future.)

Changing the `""` to `null` fix the problem. I tested locally with a patch-package.

Hoping that it fixes your problem.
